cpp/distributed/dist-mnist.cpp

```diff
@@ -45,16 +37,17 @@ struct Model : torch::nn::Module {
 };
 
 int main(int argc, char* argv[]) {
-  // MPI variables
-  int rank, numranks;
-  MPI_Init(&argc, &argv);
-  MPI_Comm_size(MPI_COMM_WORLD, &numranks);
-  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-  // end MPI variables
+  initMPIOnce();
 
+  int rank = atoi(getenv("RANK"));
+  int numranks = atoi(getenv("SIZE"));
+  c10d::ProcessGroupMPI pg(rank, numranks, MPI_COMM_WORLD);
```

Comment:
Not sure if the above is suitable to initialize the MPI environment and create the ProcessGroup API

Reply:
I think `ProcessGroupMPI::createProcessGroupMPI` would probably encompass all the initialization. It calls `initMPIOnce` for you and returns a correctly instantiated ProcessGroup

Reply:
Ok done.